src/services/authentication/hooks/index.js

```diff
@@ -73,7 +73,7 @@ const injectUserId = async (context) => {
 	const { strategy } = context.data;
 	const systemId = strategy === 'local' ? undefined : context.data.systemId;
 
-	if (strategy !== 'jwt') {
+	if (strategy !== 'jwt' && context.data.username) {
```

Comment:
If not it go to the next hook without injectUserId?

Reply:
yes, exactly. In without username, the query will produce *all* accounts and select one randomly. This is never what you want. This hook should imho only run for moodle and iserv anyway, but to refactor that is too much for the hotfix.

Reply:
You are already created a follow up ticket?